x-pack/plugins/cases/public/components/files/add_file.tsx

```diff
@@ -0,0 +1,148 @@
+import { useKibana } from '../../common/lib/kibana';
+import { casesQueriesKeys } from '../../containers/constants';
+import { useCreateAttachments } from '../../containers/use_create_attachments';
+import { CASES_FILE_KINDS } from '../../files';
+import { useCasesContext } from '../cases_context/use_cases_context';
+import * as i18n from './translations';
+
+interface AddFileProps {
+  caseId: string;
+}
+
+const AddFileComponent: React.FC<AddFileProps> = ({ caseId }) => {
+  const { owner } = useCasesContext();
+  const { client: filesClient } = useFilesContext();
+  const { notifications } = useKibana().services;
+  const queryClient = useQueryClient();
+
+  const { isLoading, createAttachments } = useCreateAttachments();
+  const [isModalVisible, setIsModalVisible] = useState(false);
+
+  const closeModal = () => setIsModalVisible(false);
+  const showModal = () => setIsModalVisible(true);
+
+  const refreshAttachmentsTable = useCallback(() => {
+    queryClient.invalidateQueries(casesQueriesKeys.caseView());
+  }, [queryClient]);
+
+  const onError = useCallback(
+    (error) => {
+      notifications.toasts.addError(error, {
+        title: i18n.FAILED_UPLOAD,
+      });
+    },
+    [notifications.toasts]
+  );
+
+  const onUploadDone = useCallback(
+    async (chosenFiles: UploadedFile[]) => {
+      if (chosenFiles.length === 0) {
+        notifications.toasts.addDanger({
+          title: i18n.FAILED_UPLOAD,
+        });
+      } else {
+        const file = chosenFiles[0];
+
+        try {
+          await createAttachments({
+            caseId,
+            caseOwner: owner[0],
+            data: [
+              {
+                type: CommentType.externalReference,
+                externalReferenceId: file.id,
+                externalReferenceStorage: {
+                  type: ExternalReferenceStorageType.savedObject,
+                  soType: FILE_SO_TYPE,
+                },
+                externalReferenceAttachmentTypeId: FILE_ATTACHMENT_TYPE,
+                externalReferenceMetadata: {
+                  files: [
+                    {
+                      name: file.fileJSON.name,
+                      extension: file.fileJSON.extension ?? '',
+                      mimeType: file.fileJSON.mimeType ?? '',
+                      createdAt: file.fileJSON.created,
+                    },
+                  ],
+                },
+              },
+            ],
+            updateCase: refreshAttachmentsTable,
+            throwOnError: true,
+          });
+
+          notifications.toasts.addSuccess({
+            title: i18n.SUCCESSFUL_UPLOAD,
+            text: i18n.SUCCESSFUL_UPLOAD_FILE_NAME(file.fileJSON.name),
+          });
+        } catch (error) {
+          // error toast is handled inside  createAttachments
+
+          // we need to delete the file if attachment creation failed
+          await filesClient.delete({ kind: CASES_FILE_KINDS[APP_ID].id, id: file.id });
```

Comment:
Good catch. Yep: https://github.com/elastic/kibana/blob/main/x-pack/plugins/cases/common/constants/files.ts#L16

One other note here. We'll actually need to call the cases file attachment delete API here instead because my PR will be disabling the file service HTTP delete API. I'll make sure the deletion implementation handles the situation where a fileId will exist but the case attachment may not just to be safe.

Reply:
Ah, Initially I thought it was the same because in `x-pack/plugins/cases/public/files/index.ts` we have:
```
const buildFileKind = (owner: Owner): FileKindBrowser => {
  return {
    id: constructFileKindIdByOwner(owner),
    allowedMimeTypes: ALLOWED_MIME_TYPES,
    maxSizeBytes: MAX_FILE_SIZE,
  };
};

/**
 * The file kind definition for interacting with the file service for the UI
 */
export const CASES_FILE_KINDS: Record<Owner, FileKindBrowser> = {
  [APP_ID]: buildFileKind(APP_ID),
  [SECURITY_SOLUTION_OWNER]: buildFileKind(SECURITY_SOLUTION_OWNER),
  [OBSERVABILITY_OWNER]: buildFileKind(OBSERVABILITY_OWNER),
};
```
Where the `.id` already returns the result of passing the "owner" to the function you specified @jonathan-buttner .

But now I guess it should be this instead, right?
```
  const { owner } = useCasesContext();
  ...
  await filesClient.delete({ kind: constructFileKindIdByOwner(owner), id: file.id });
```

Reply:
> But now I guess it should be this instead, right?

Yep!